service/history/MockProcessor.go

```diff
@@ -30,16 +30,24 @@ type MockProcessor struct {
 }
 
 // process is mock implementation for process of Processor
-func (_m *MockProcessor) process(task queueTaskInfo) error {
+func (_m *MockProcessor) process(task queueTaskInfo) (int, error) {
```

Comment:
What does the int return value signify? What outcome does process other than it worked or error?

Reply:
this is to some degree an anti pattern, but can be useful for this use case.

basically the scope is unknown, before processing the task. we can either do a switch case at the top layer to get the scope then internal implementation do an switch case again to redirect the task to target processing logic, or just let the target processing logic return the scope, which eliminate unnecessary logic.